swarm/network/stream/delivery.go

```diff
@@ -173,12 +174,48 @@ func (d *Delivery) Close() {
 
 // FindPeer is returning the closest peer from Kademlia that a chunk
 // request hasn't already been sent to
-func (d *Delivery) FindPeer(req *storage.Request) (*Peer, error) {
+func (d *Delivery) FindPeer(ctx context.Context, req *storage.Request) (*Peer, error) {
 	var sp *Peer
 	var err error
 
+	osp, _ := ctx.Value("remote.fetchh").(opentracing.Span)
+
 	depth := d.kad.NeighbourhoodDepth()
 
+	originPo := -1
+	d.kad.EachConn(req.Addr[:], 255, func(p *network.Peer, po int) bool {
+		id := p.ID()
+
+		// get po between chunk and origin
+		if req.Origin.String() == id.String() {
+			originPo = po
+			return false
+		}
+
+		return true
+	})
+
+	if osp != nil {
+		osp.LogFields(olog.Int("originPo", originPo))
+		osp.LogFields(olog.Int("depth", depth))
+	}
+
+	selectedPeerPo := -1
+
+	myPo := chunk.Proximity(req.Addr, d.kad.BaseAddr())
+
+	if osp != nil {
+		osp.LogFields(olog.Int("myPo", myPo))
+	}
+
+	skipped := 0
+
+	// do not forward requests if origin proximity is bigger than our node's proximity
```

Comment:
This condition is independent of the `EachConn` iteration and should be moved above ~line 184